### PR TITLE
feat: Optimize post page

### PR DESCRIPTION
**Pull request by AmarTrebinjac**
Changed post content types to dynamic imports. Resulted in a significant performance increase.

Preview LEFT, Production RIGHT:

details_preview_production

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

WT-2208

### Preview domain
https://wt-2208.preview.app.daily.dev

### CODE REVIEW

File: packages/webapp/__tests__/setup.ts [hunk 1]
```diff
@@ -7,20 +7,7 @@ process.env.NEXT_PUBLIC_API_URL = 'http://localhost:3000';
 process.env.NEXT_PUBLIC_WEBAPP_URL = '/';
 
 /* eslint-disable @typescript-eslint/no-explicit-any */
-
-jest.mock('next/dynamic', () => (func: () => Promise<any>) => {
-  let component: any = null;
-  func().then((module: any) => {
-    component = module.default;
```
Comment (AmarTrebinjac) [inline packages/webapp/__tests__/setup.ts:14]: This caused a bazillion errors due to it expecting the dynamic import to be default:

notAllDynamicAreDefault

But not all components we want to import dynamically are necessarily exported as a default.

Removing this made all tests pass, except one, which passed when I wrapped it in an act. So it didn't seem to be really necessary?

The code was written 4 years ago, so maybe the recent updates we've done to NextJs made it redundant

File: packages/webapp/pages/posts/[id]/index.tsx [hunk 2]
```diff
@@ -87,16 +110,10 @@ const PostPage = ({ id, initialData }: Props): ReactElement => {
     [PostType.Share, PostType.Welcome, PostType.Freeform].includes(post?.type),
     featureTheme && 'bg-transparent',
   );
-  const seoTitle = () => {
-    if (post?.type === PostType.Share && post?.title === null) {
-      return `Shared post at ${post?.source?.name}`;
-    }
```
Comment (AmarTrebinjac) [inline packages/webapp/pages/posts/[id]/index.tsx:93]: Moved the function outside the component so it isn't recreated every render, and follows best practices

File: packages/webapp/pages/posts/[id]/index.tsx [hunk 3]
```diff
@@ -107,8 +124,6 @@ const PostPage = ({ id, initialData }: Props): ReactElement => {
     },
   };
 
-  const seoComponent = <NextSeo {...seo} />;
```
Comment (sshanzel) [inline packages/webapp/pages/posts/[id]/index.tsx:110]: Any particular reason why we preferred rendering then spread on both areas?

**Review by sshanzel — APPROVED** [review]
Don't have much to add 🚀 

Just some questions for my understanding 🕵️

**Review by nensidosari — APPROVED** [review]

**Comment by nensidosari** [comment]
Btw would suggest u add the scores as well in the images (or just add the links) - I tested and seems to be worse 😬 

https://pagespeed.web.dev/analysis/https-app-daily-dev-posts-react-spa-in-astro-with-tanstack-router-j27qrczi2/vci95js4ic?form_factor=mobile

https://pagespeed.web.dev/analysis/https-wt-2208-preview-app-daily-dev-posts-react-spa-in-astro-with-tanstack-router-j27qrczi2/ps101io7kn?form_factor=mobile

**Comment by AmarTrebinjac** [comment]
> Btw would suggest u add the scores as well in the images (or just add the links) - I tested and seems to be worse 😬
> 
> https://pagespeed.web.dev/analysis/https-app-daily-dev-posts-react-spa-in-astro-with-tanstack-router-j27qrczi2/vci95js4ic?form_factor=mobile
> 
> https://pagespeed.web.dev/analysis/https-wt-2208-preview-app-daily-dev-posts-react-spa-in-astro-with-tanstack-router-j27qrczi2/ps101io7kn?form_factor=mobile

@nensidosari 

I'll add the links next time!

The socres vary a lot depending on latency and other issues you can't control, that's why the only reliable thing to look at are the numbers in the details (They even say so on the page). I ran one more test and here the preview wins by a lot:

https://pagespeed.web.dev/analysis/https-app-daily-dev-posts-react-spa-in-astro-with-tanstack-router-j27qrczi2/qt7g5e10od?form_factor=mobile

https://pagespeed.web.dev/analysis/https-wt-2208-preview-app-daily-dev-posts-react-spa-in-astro-with-tanstack-router-j27qrczi2/sa2ytk5kr3?form_factor=mobile

**Pull request merged**